Camera2Extensions/app/src/main/java/com/example/android/camera2/extensions/fragments/CameraFragment.kt

```diff
@@ -722,14 +725,26 @@ class CameraFragment : Fragment(), TextureView.SurfaceTextureListener {
    * Creates the still image reader and sets up OnImageAvailableListener
    */
   private fun createStillImageReader(): ImageReader {
+    var stillFormat: Int
+    var stillCaptureSize: Size
+
     val yuvColorEncodingSystemSizes = extensionCharacteristics.getExtensionSupportedSizes(
       currentExtension, ImageFormat.YUV_420_888
     )
     val jpegSizes = extensionCharacteristics.getExtensionSupportedSizes(
       currentExtension, ImageFormat.JPEG
     )
-    val stillFormat = if (jpegSizes.isEmpty()) ImageFormat.YUV_420_888 else ImageFormat.JPEG
-    val stillCaptureSize = if (jpegSizes.isEmpty()) yuvColorEncodingSystemSizes[0] else jpegSizes[0]
+    stillFormat = if (jpegSizes.isEmpty()) ImageFormat.YUV_420_888 else ImageFormat.JPEG
+    stillCaptureSize = if (jpegSizes.isEmpty()) yuvColorEncodingSystemSizes[0] else jpegSizes[0]
+
+    val jpegRSizes = extensionCharacteristics.getExtensionSupportedSizes(
+      currentExtension, ImageFormat.JPEG_R
+    )
+    if (args.hdr) {
```

Comment:
nit: we can simplify this a bit to `if (args.hdr && jpegSizes.isNotEmpty())`

Reply:
Done

Reply:
Done